engine/equations.cc

```diff
@@ -641,9 +641,16 @@ namespace MathDAG
     shared_ptr<VariableDAG> r(new VariableDAG(valueId, nm, type));
     expressionCache.insert(valueId, r);
     r->init=vv->init;
-    if (auto v=minsky.definingVar(valueId))
+    if (auto v=minsky.definingVar(valueId)) {
       if (v->type()!=VariableType::integral && v->numPorts()>1 && !v->ports[1]->wires().empty())
         r->rhs=getNodeFromWire(*v->ports[1]->wires()[0]);
+	  // ensure that flows used as intial conditions, which inherit their values from other vars, correctly initialise corresponding stock var. for ticket 1137.
+	  auto vi=minsky.variableValues[VariableValue::valueId(v->group.lock(),vv->init)];
+	  if (vi->lhs()) {
+		  vv->init=str(vv->value());        
+		  r->init=vv->init;
```

Comment:
This then overwrites the initialisation string with the current value of the defining variable. This is definitely the wrong this to do.

Reply:
Okay, thanks